keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/binary/DoubleGreaterThanMaskVertex.java

```diff
@@ -0,0 +1,22 @@
+package io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.binary;
+
+import io.improbable.keanu.annotation.ExportVertexToPythonBindings;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.LoadVertexParam;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+
+import static io.improbable.keanu.tensor.TensorShapeValidation.checkHasOneNonLengthOneShapeOrAllLengthOne;
+
+public class DoubleGreaterThanMaskVertex extends DoubleBinaryOpVertex {
+    @ExportVertexToPythonBindings
+    public DoubleGreaterThanMaskVertex(@LoadVertexParam(LEFT_NAME) DoubleVertex left,
+                                       @LoadVertexParam(RIGHT_NAME) DoubleVertex right) {
+        super(checkHasOneNonLengthOneShapeOrAllLengthOne(left.getShape(), right.getShape()), left, right);
```

Comment:
You do the shape check here but not in DoubleGreaterThanOrEqualToMaskVertex. Is there a reason that this is a special case?

Reply:
Good eye. The mistake is actually doing the check in the first place. `DoubleBinaryOpVertex` does the shape check already.